src/Microsoft.Performance.SDK.Runtime/Extensibility/FilteredDataRetrieval.cs

```diff
@@ -102,6 +126,11 @@ private void ValidateSourceDataCooker(DataOutputPath dataOutputPath)
             }
         }
 
+        private bool TryValidateSourceDataCooker(DataOutputPath dataOutputPath)
+        {
+            return this.extensionDependencies.RequiredSourceDataCookerPaths.Contains(dataOutputPath.CookerPath);
```

Comment:
do we want to avoid using `.CookerPath`?

Reply:
I don't know, is there?

Reply:
I was just thinking if we're trying to use the rep of a cooker path less maybe we should be passing around/using the actual DataOutPath objects instead. We'd could just implement IComparable/IEquatable on the paths

Reply:
This isn't a string, it's a DataCookerPath, which does implement IEquatable<>. And We want to compare the cooker paths, not the DataOutputPath, which is why we're using that property.

Reply:
Ah, my bad. I thought this DataCookerPath.CookerPath which was the string.